modules/rostests/apitests/user32/PrivateExtractIcons.c

```diff
@@ -3,41 +3,128 @@
  * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
  * PURPOSE:         Test for PrivateExtractIcons
  * PROGRAMMER:      Hermes Belusca-Maito
+ *                  Doug Lyons <douglyons@douglyons.com>
  */
 
 #include "precomp.h"
+#include <stdio.h>
+
+BOOL FileExists(LPCSTR FileName)
+{
+    FILE *fp = NULL;
+
+    fp = fopen(FileName, "r");
+    if (fp != NULL)
+    {
+        fclose(fp);
+        return TRUE;
+    }
+    return FALSE;
+}
```

Comment:
What about `PathFileExistsW`? [[👉 MSDN]](https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathfileexistsw) Also, SO recommends `GetFileAttributes` that doesn't require `shlwapi.dll`: https://stackoverflow.com/a/6218957

Reply:
Done with GetFileAttributes. Thanks.